File: hammer-adventure/src/main/java/dev/vini2003/hammer/adventure/HA.java
package dev.vini2003.hammer.adventure;

import dev.vini2003.hammer.adventure.registry.common.HAEvents;
import net.fabricmc.api.ModInitializer;
import net.kyori.adventure.platform.fabric.FabricServerAudiences;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Internal
public class HA implements ModInitializer {
	private static FabricServerAudiences audiences;
	
	public static FabricServerAudiences getAudiences() {
		return audiences;
	}
	
	public static void setAudiences(FabricServerAudiences audiences) {
		HA.audiences = audiences;
	}
	
	@Override
	public void onInitialize() {
		HAEvents.init();
	}
}

File: hammer-adventure/src/main/kotlin/dev/vini2003/hammer/adventure/HA.kt
package dev.vini2003.hammer.adventure

import dev.vini2003.hammer.adventure.common.util.AdventureUtils
import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents
import net.kyori.adventure.platform.fabric.FabricServerAudiences

object HA : ModInitializer {
	private var ADVENTURE_INSTANCE: FabricServerAudiences? = null
	
	@JvmStatic
	@get:JvmName("getAudience")
	val ADVENTURE
		get() = ADVENTURE_INSTANCE!!
	
	override fun onInitialize() {
		ServerLifecycleEvents.SERVER_STARTING.register { server ->
			ADVENTURE_INSTANCE = AdventureUtils.createAudience(server)
		}
		
		ServerLifecycleEvents.SERVER_STOPPED.register {
			ADVENTURE_INSTANCE = null
		}
	}
	
}